app/src/main/java/com/duckduckgo/app/browser/omnibar/QueryUrlConverter.kt

```diff
@@ -28,8 +28,13 @@ import javax.inject.Inject
 
 class QueryUrlConverter @Inject constructor(private val requestRewriter: RequestRewriter) : OmnibarEntryConverter {
 
-    override fun convertQueryToUrl(searchQuery: String, vertical: String?): String {
-        if (UriString.isWebUrl(searchQuery)) {
+    override fun convertQueryToUrl(searchQuery: String, vertical: String?, queryOrigin: QueryOrigin): String {
+        val isUrl = when (queryOrigin) {
+            is QueryOrigin.FromAutocomplete -> queryOrigin.nav ?: UriString.isWebUrl(searchQuery)
```

Comment:
We have this logic twice: `queryOrigin.nav ?: UriString.isWebUrl(searchQuery)`; here and in `getAutoCompleteSearchResults`

I haven't thought about it too much here, but it feels like we'd able to have this logic happen only once and by the time we get to here we wouldn't have to do it again.

is not something to block on, and it might be necessary after all, so just posing the question if both are needed

Reply:
That is correct! I've removed that code and tested it, everything works as expected.